### PR TITLE
feat: alternate log location support - InfluxDB2

### PR DESCRIPTION
**Pull request by Jellayy**
This change adds alternative log location support to FindMyHistory through the `--log_location` flag. This change allows the addition of any number of custom log locations by updating `LogManager._save_log()` to route logs to the currently selected destination.

I've built out one logger for InfluxDB2 using the influxdb-client library. InfluxDB2 easily connects with Grafana, allowing you to create dashboards using the [Geomap Visualization](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/geomap/#route-layer-beta) to plot your historic FindMy location data on a map. README documentation has been updated with steps for usage.

Default behavior is unchanged, running `main.py` alone will continue to log to a csv logfile.

### CODE REVIEW

File: lib/log_manager.py [hunk 1]
```diff
@@ -3,22 +3,37 @@
...
+            self._no_date_folder = no_date_folder
+        elif self._log_location == 'influx':
+            self._influx_org = influx_org
+            self._influx_bucket = influx_bucket
+            influx_client = InfluxDBClient(
+                url=influx_host,
+                token=influx_token,
+                org=self._influx_org
+            )
+            self._influx_write_api = influx_client.write_api()
```
Comment (fjxmlzn) [inline lib/log_manager.py:36]: Thanks so much for the nice feature! Could you add an else statement here to raise an error when self._log_location is neither local/influx?

**Review by fjxmlzn — CHANGES_REQUESTED** [review]
Thank you very much for the nice feature! I added two minor comments. Could you help make the change? 

Thank you.

File: lib/log_manager.py (hunk 1 above)
Comment (Jellayy) [reply lib/log_manager.py:36]: Updated to raise a ValueError on an unsupported log location

File: main.py [hunk 2]
```diff
@@ -68,8 +68,53 @@ def parse_args():
         action='store_true',
         help='By default, the logs of each day will be saved in a separated '
              'folder. Use this option to turn it off.')
+    parser.add_argument(
+        '--log_location',
+        type=str,
+        action='store',
+        default='local',
+        help='Location to log findmy data. Default: local'
```
[the remark this answers is not in the document]
Comment (Jellayy) [reply main.py:80]: Updated with 'local' and 'influx' as choices

**Comment by Jellayy** [comment]
> Thank you very much for the nice feature! I added two minor comments. Could you help make the change?
> 
> Thank you.

Got em in!

**Review by fjxmlzn — APPROVED** [review]

**Pull request merged**